cep-20.2.md

```diff
@@ -0,0 +1,650 @@
+# a free form dictionary
+extra:
+  <key>: <value>
+```
+
+## Example recipe
+
+What follows is an example recipe, using the YAML syntax discussed in https://github.com/conda-incubator/ceps/pull/54
+
+```yaml
+context:
+  name: xtensor
+  version: 0.24.6
+
+package:
+  name: ${{ name|lower }}
+  version: ${{ version }}
+
+source:
+  url: https://github.com/xtensor-stack/xtensor/archive/${{ version }}.tar.gz
+  sha256: f87259b51aabafdd1183947747edfff4cff75d55375334f2e81cee6dc68ef655
+
+build:
+  number: 0
+  skip:
+    # note that the value is a minijinja expression
+    - osx or win
+
+requirements:
+  build:
+    - ${{ compiler('cxx') }}
+    - cmake
+    - if: unix
+      value: make
+  host:
+    - xtl >=0.7,<0.8
+  run:
+    - xtl >=0.7,<0.8
+  run_constrained:
+    - xsimd >=8.0.3,<10
+
+test:
+  - script:
+      - if: unix
+        value:
+          - test -d ${PREFIX}/include/xtensor
+          - test -f ${PREFIX}/include/xtensor/xarray.hpp
+          - test -f ${PREFIX}/share/cmake/xtensor/xtensorConfig.cmake
+          - test -f ${PREFIX}/share/cmake/xtensor/xtensorConfigVersion.cmake
+      - if: win
+        value:
+          - if not exist %LIBRARY_PREFIX%\include\xtensor\xarray.hpp (exit 1)
+          - if not exist %LIBRARY_PREFIX%\share\cmake\xtensor\xtensorConfig.cmake (exit 1)
+          - if not exist %LIBRARY_PREFIX%\share\cmake\xtensor\xtensorConfigVersion.cmake (exit 1)
+  # compile a test package
+  - if: unix
+    then:
+      files:
+        - testfiles/cmake/*
+      extra_requirements:
+        build:
+          - ${{ compiler('cxx') }}
+          - cmake
+          - ninja
+      script: |
+        cd testiles/cmake/
+        mkdir build; cd build
+        cmake -GNinja ..
+        cmake --build .
+        ./target/hello_world
+
+  - downstream: xtensor-python
+  - downstream: xtensor-blas
+  - imports:
+      - xtensor_python
+      - xtensor_python.numpy_adapter
+```
+
+Or a multi-output package
+
+```yaml
+context:
+  name: mamba
+  libmamba_version: "1.4.2"
+  libmambapy_version: "1.4.2"
+  # can also reference previous variables here
+  mamba_version: ${{ libmamba_version }}
+  release: "2023.04.06"
+  libmamba_version_split: ${{ libmamba_version.split('.') }}
+
+# we can leave this out
+# package:
+#   name: mamba-split
+
+# this is inherited by every output
+source:
+  url: https://github.com/mamba-org/mamba/archive/refs/tags/${{ release }}.tar.gz
+  sha256: bc1ec3de0dd8398fcc6f524e6607d9d8f6dfeeedb2208ebe0f2070c8fd8fdd83
+
+build:
+  number: 0
+
+outputs:
+  - package:
+      name: libmamba
+      version: ${{ libmamba_version }}
+
+    build:
+      script: ${{ "build_mamba.sh" if unix else "build_mamba.bat" }}
+      run_exports:
+        - ${{ pin_subpackage('libmamba', max_pin='x.x') }}
+      ignore_run_exports:
+        - spdlog
+        - if: win
+          then: python
+
+    requirements:
+      build:
+        - ${{ compiler('cxx') }}
+        - cmake
+        - ninja
+        - ${{ "python" if win }}
```

Comment:
What's the returned value happens when there's no else clause and the if clause is not true? `null`?

Reply:
Yes, a `Yaml::Null` value (that we filter out in these `ConditionalList`).